test-data/unit/check-semanal-error.test

```diff
@@ -76,11 +76,20 @@ break # E: "break" outside loop
 continue # E: "continue" outside loop
 
 [case testYieldOutsideFunction]
```

Comment:
I think all the yield/yield from cases can go in a single test case. Having more test cases hurts the performance of the test suite.

Also, could you test some more cases:
- yield in the outermost comprehension (which is valid, because it's executed in the enclosing function's scope)
- yield in an inner comprehension or in the `if` clause

Reply:
Added tests -- is that what you had in mind?